dGame/dComponents/TriggerComponent.cpp

```diff
@@ -0,0 +1,249 @@
+			// TODO:
+			// CastSkill 	[skillID]
+		} else if (id == "displayZoneSummary") {
+			// TODO:
+			// displayZoneSummary 	[1 for zone start, 0 for zone end]
+		} else if (id == "SetPhysicsVolumeStatus") {
+			// TODO:
+			// SetPhysicsVolumeStatus 	[“On”, “Off”]
+		} else if (id == "setModelToBuild") {
+			// TODO:
+			// setModelToBuild 	[template ID]
+		} else if (id == "spawnModelBricks") {
+			// TODO:
+			// spawnModelBricks 	[amount, from 0 to 1],[x],[y],[z]
+		} else if (id == "ActivateSpawnerNetwork") {
+			// TODO:
+			// ActivateSpawnerNetwork 	[Spawner Network Name]
+		} else if (id == "DeactivateSpawnerNetwork") {
+			// TODO:
+			// DeactivateSpawnerNetwork 	[Spawner Network Name]
+		} else if (id == "ResetSpawnerNetwork") {
+			// TODO:
+			// ResetSpawnerNetwork 	[Spawner Network Name]
+		} else if (id == "DestroySpawnerNetworkObjects") {
+			// TODO:
+			// DestroySpawnerNetworkObjects 	[Spawner Network Name]
+		} else if (id == "Go_To_Waypoint") {
+			// TODO:
+			// Go_To_Waypoint 	[Waypoint index],(“true” to allow direction change, otherwise “false”),(“true” to stop at waypoint, otherwise “false”)
+		} else if (id == "ActivatePhysics") {
+			// TODO:
+			// ActivatePhysics 	“true” to activate and add to world, “false” to deactivate and remove from the world
+		} else {
+			Game::logger->LogDebug("TriggerComponent", "Trigger Event %s does not exist!", id.c_str());
+		}
+	}
+}
+
+std::vector<std::string> TriggerComponent::ParseArgs(std::string args){
+	std::vector<std::string> argArray;
+	std::stringstream ssData(args);
+	std::string token;
+	char deliminator = ',';
+
+	while (std::getline(ssData, token, deliminator)) {
+		std::string lowerToken;
+		for (char character : token) {
+			lowerToken.push_back(std::tolower(character)); // make lowercase to ensure it works
+		}
+		argArray.push_back(lowerToken);
+	}
+	return argArray;
+}
+
+void TriggerComponent::HandleSetPhysicsVolume(Entity* targetEntity, std::vector<std::string> argArray, std::string target) {
+	PhantomPhysicsComponent* phanPhys = m_Parent->GetComponent<PhantomPhysicsComponent>();
+	if (!phanPhys) return;
+
+	phanPhys->SetPhysicsEffectActive(true);
+	uint32_t effectType = 0;
+	if (argArray[0] == "push") effectType = 0;
+	else if (argArray[0] == "attract") effectType = 1;
+	else if (argArray[0] == "repulse") effectType = 2;
+	else if (argArray[0] == "gravity") effectType = 3;
+	else if (argArray[0] == "friction") effectType = 4;
+
+	phanPhys->SetEffectType(effectType);
+	phanPhys->SetDirectionalMultiplier(std::stof(argArray[1]));
+	if (argArray.size() > 4) {
+		NiPoint3 direction = NiPoint3::ZERO;
+		GeneralUtils::TryParse<float>(argArray[2], direction.x);
+		GeneralUtils::TryParse<float>(argArray[3], direction.y);
+		GeneralUtils::TryParse<float>(argArray[4], direction.z);
+		phanPhys->SetDirection(direction);
+	}
+	if (argArray.size() > 5) {
+		phanPhys->SetMin(std::stoi(argArray[6]));
+		phanPhys->SetMax(std::stoi(argArray[7]));
```

Comment:
For sanity TryParse might be good to have

Reply:
will do, just copy pasted what was there originally